File: Sample codes/Get Request Status/PHP/get_request_status.php
<?php
$url="https://api.shuftipro.com/status";
$post_data = array(
	"client_id"           => "Your client id provided by Shuftipro", 
	"reference"           => "Your unique request reference",
	);
ksort($post_data);//Sort the all request parameter.
$raw_data = implode("", $post_data) . "YOUR_SECRET_KEY"; //Replace with your secret key provided by the Shuftipro;
$signature              = hash("sha256", $raw_data);
$post_data["signature"] = $signature;
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $url); curl_setopt($ch, CURLOPT_POST, 1); curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data); curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$response = curl_exec($ch);
curl_close($ch);
?>

File: Sample codes/Get Request Status/Python/get_request_status.py
import hashlib 
import requests 
import  json
import collections

url = 'https://api.shuftipro.com/status'
CLIENT_ID = 'Your client id provided by Shuftipro'
SECRET_KEY = 'YOUR_SECRET_KEY'

post_data = {
"client_id"           : CLIENT_ID, 
"reference"           : "Your unique request reference", 
}


post_data = collections.OrderedDict(sorted(post_data.items())) #sort the dictionary raw_data = "".join(post_data.values()) + SECRET_KEY #get values from dictionary and append secret key

hash_object = hashlib.sha256(raw_data) #calculating sha 256 hash signature = hash_object.hexdigest()

post_data['signature'] = signature #append signature to data dictionary response = requests.post(url, post_data).json() #send POST request to API
